src/MongoDB.Driver.Core/Core/Compression/CompressorTypeMapper.cs

```diff
@@ -0,0 +1,51 @@
+﻿/* Copyright 2020–present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+namespace MongoDB.Driver.Core.Compression
+{
+    internal static class CompressorTypeMapper
+    {
+        public static string ToServerName(CompressorType compressorType)
+        {
+            switch (compressorType)
+            {
+                case CompressorType.Noop:
+                    return "noop";
+                case CompressorType.Zlib:
+                    return "zlib";
+                case CompressorType.Snappy:
+                    return "snappy";
+                case CompressorType.ZStandard:
+                    return "zstd";
+                default:
+                    return compressorType.ToString().ToLowerInvariant();
```

Comment:
I think you should throw an exception.

CompressorType is an enum and if we don't have a case for all the valid enum values it is OUR bug.

Reply:
The spec says about simialr situation:

>If the server has no compression algorithms in common with the client, it sends back an isMaster response without a compression field. Clients MAY issue a log level event to inform the user, but MUST NOT error.

I agree that it looks like a bug if we didn't add a value to the enum. But I stil think that we should not throw the error here

Reply:
and rely on the server behavior